### PR TITLE
Inhibit alerts for nodes in maintenance

### PR DESCRIPTION
**Pull request by auhlig**
This PR introduces a new inhibition rule to prevent alerts for nodes in maintenance.
This is opt-in, by setting the label `inhibited-by: node-maintenance` on the alert that should be inhibited.

### CODE REVIEW

File: prometheus-rules/prometheus-kubernetes-rules/alerts/maintenance.alerts.tpl [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    for: 15m
+    labels:
+      tier: {{ required ".Values.tier missing" .Values.tier }}
+      service: node
+      severity: none
+      context: node
+      meta: "Node {{`{{ $labels.node }}`}} is in maintenance."
+    annotations:
+      summary: Node in maintenance
+      description: "Node {{`{{ $labels.node }}`}} is in scheduled maintenance. This inhibits other alerts for the duration of the maintenance."
```
Comment (databus23) [inline prometheus-rules/prometheus-kubernetes-rules/alerts/maintenance.alerts.tpl:17]:
```suggestion
      description: "Node {{`{{ $labels.node }}`}} is in scheduled maintenance. Add the label `inhibited-by: node-maintenance` to alerts that should be inhibited while a node is in maintenance"
```

**Review by databus23 — COMMENTED** [review]
A few minor suggestions

File: prometheus-rules/prometheus-kubernetes-rules/alerts/kubelet.alerts.tpl [hunk 2]
```diff
@@ -12,6 +12,7 @@ groups:
       context: kubelet
       dashboard: kubernetes-health
       playbook: docs/support/playbook/kubernetes/k8s_node_not_ready.html
+      inhibited-by: node-maintenance
```
[the remark this answers is not in the document]
Comment (auhlig) [reply prometheus-rules/prometheus-kubernetes-rules/alerts/kubelet.alerts.tpl:15]: Ah. True. Actually, we don't achieve anything in this case as the `ManyKubeletDown` inhibits the `KubeletDown` alert.
Mh. We might need to exclude nodes in maintenance from this on metric level :-/

[the remark this answers is not in the document]
Comment (databus23) [reply prometheus-rules/prometheus-kubernetes-rules/alerts/kubelet.alerts.tpl:15]: Yes I agree

**Comment by databus23** [comment]
I modified the queries to fix the Many* alerts wdyt?

**Comment by auhlig** [comment]
LGTM

**Pull request merged**